File: 221110/rescueBoat.test.js
import solution from './rescueBoat';

test('solution case 1', () => {
  const weights = [70, 50, 80, 50];
  const limit = 100;
  expect(solution(weights, limit)).toBe(3);
});

test('solution case 2', () => {
  const weights = [70, 50, 80];
  const limit = 100;
  expect(solution(weights, limit)).toBe(3);
});

File: 221109/RescueBoatTest.java
import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

class RescueBoatTest {
    @Test
    void solutionCase1() {
        RescueBoat test = new RescueBoat();

        int[] weights = new int[]{70, 50, 80, 50};
        int limit = 100;
        assertThat(test.solution(weights, limit)).isEqualTo(3);
    }

    @Test
    void solutionCase2() {
        RescueBoat test = new RescueBoat();

        int[] weights = new int[]{70, 50, 80};
        int limit = 100;
        assertThat(test.solution(weights, limit)).isEqualTo(3);
    }
}
